mso/resource_mso_schema_site_external_epg.go

```diff
@@ -156,14 +190,19 @@ func resourceMSOSchemaSiteExternalEpgCreate(d *schema.ResourceData, m interface{
 		if err != nil {
 			return err
 		}
-
+		
 		l3outRefMap := make(map[string]interface{})
 
-		l3outRefMap["schemaId"] = schemaId
-		l3outRefMap["templateName"] = templateName
-		l3outRefMap["l3outName"] = l3outName
+		if l3outOnApic {
+			siteEpgMap["l3outRef"] = ""
+		} else {		
+			l3outRefMap["schemaId"] = l3outSchema
+			l3outRefMap["templateName"] = l3outTemplate
+			l3outRefMap["l3outName"] = l3outName
```

Comment:
I was thinking we still miss a thing. We can do two things actually.

- We should do a check that the template and schema are provided when they are optional since I believe they are required for the ref to be constructed.
- Or we can set the schema/template of the external epg when these are not provided. ( my preference )

Another thing is should we add validation of config to make  l3outSchema/l3outTemplate mutually exclusive from l3out_on_apic?

Reply:
This behaviour should also be described in documentation.